misc/snapshotter/snapshotter.sh

```diff
@@ -217,11 +217,11 @@ function cleanup_snapshotter() {
     fi
     wait_service_active 30 5 ${CONTAINER_RUNTIME}
     echo "Removing nydus-snapshotter artifacts from host"
-    rm -f "${SNAPSHOTTER_BINARY}"
-    rm -f "${NYDUS_BINARY_DIR}/nydus*"
-    rm -rf "${NYDUS_CONFIG_DIR}/*"
-    rm -rf "${SNAPSHOTTER_SCRYPT_DIR}/*"
-    rm -rf "${NYDUS_LIB_DIR}/*"
+    bash -c "rm -f ${SNAPSHOTTER_BINARY}"
+    bash -c "rm -f ${NYDUS_BINARY_DIR}/nydus*"
+    bash -c "rm -rf ${NYDUS_CONFIG_DIR}/*"
+    bash -c "rm -rf ${SNAPSHOTTER_SCRYPT_DIR}/*"
+    bash -c "rm -rf ${NYDUS_LIB_DIR}/*"
```

Comment:
This doesn't change anything with respect to pathname expansion : `${NYDUS_LIB_DIR}/*` is expanded all the same above and in the original code. The `Regex not evaluated in the rm statement.` sentence in the description is thus incorrect.

The problem is that the result of the expansion is passed to `rm` as a single argument. This causes `rm` to look for an file that doesn't exist if this argument contains several items, e.g. `/path/to/nydus/lib/dir/FOO /path/to/nydus/lib/dir/BAR`.

Also, `bash -c ""` doesn't buy anything here, you could just make it :
```suggestion
    rm -rf ${NYDUS_LIB_DIR}/*
```

Reply:
Even this command wasn't working during my testing. It had to do with the fact that the script is required to run as sudo.

I found this link, which I think helps explain the issue:
https://superuser.com/questions/980472/remove-files-from-a-root-owned-directory-using-sudo-with-wildcards

Reply:
The link above would explain the issue if you needed to do something like :

```
    sudo bash -c "rm -rf ${NYDUS_LIB_DIR}/*"
```

as the top level shell running the script might not be able to access the content of the `${NYDUS_LIB_DIR}/` directory.

If the script itself is run with `sudo` then it will be able to access the directory content and honor the pathname expansion.

Reply:
Think I understand now. Fixed it.